coopiteasy_custom/models/account_analytic_line.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.model
+    def reset_so_line(self):
+        lines = self.search([("so_line.order_id.state", "=", "sale")])
+        for line in lines:
+            vals = {
+                "product_uom_id": line.product_uom_id.id,
+                "amount": -0.0,
+                "account_id": line.account_id.id,
+            }
+            res = line.sudo()._get_sale_order_line(vals=vals)
+            line.write(vals)
+            print(res)
```

Comment:
Do logging instead?

Reply:
The answer is of course yes but : this code and cron was used to fix some badly configured projects. I think it should be removed altogether in version 12. Maybe check with analysts.